File: 1C/euklides.py
def nwd_v1(a, b):
    licznik = 0
    while a != b:
        licznik += 1
        if a > b:
            a = a - b
        else:
            b = b - a
    print("Powtórzeń:", licznik)
    print(a)


def nwd_v2(a, b):
    licznik = 0
    while a > 0:
        licznik += 1
        a = a % b
        b -= a
    print("Powtórzeń:", licznik)
    print(b)


def main():
    a = int(input("Podaj liczbę: "))
    b = int(input("Podaj liczbę: "))
    nwd_v1(a, b)
    nwd_v2(a, b)
    return 0


main()

File: 2BP4/cpp_1/euklides.cpp
/*
 * euklides.cpp
 */
#include <iostream>
using namespace std;

int pobierzLiczbe() {
    int liczba;
    cout << "Podaj liczbę: " << endl;
    cin >> liczba;
    return liczba;
}

int main(int argc, char **argv)
{
    int a, b;
    a = pobierzLiczbe();
    b = pobierzLiczbe();
    cout << "NWD(" << a << ", " << b << ") = ";
    int licznik = 0;
    while (a != b) {
        if (a > b) {
            a -= b;
        } else {
            b -= a;
        }
        licznik++;
    }
    cout << a << endl;
    cout << "Powtórzeń: " << licznik;
    
    return 0;
}

